src/utils/work_unit.py

```
@@ -84,16 +97,17 @@ def parse_work_unit_key(work_unit_key: str) -> ParsedWorkUnit:
         )
 
     if task_type == "dream":
-        if len(parts) != 4:
+        if len(parts) != 5:
             raise ValueError(
                 f"Invalid work_unit_key format for task_type {task_type}: {work_unit_key}"
             )
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)